### PR TITLE
Cam init

### PR DESCRIPTION
**Pull request by mfdeakin-sandia**
This brings us super close to working in CAM - there don't appear to be any memory related seg faults left after the ones here are merged, only floating point issues from some fields not being initialized properly I think. Suspects include `hybrid_ai_delta`, `hybrid_bi_delta`, and `ps_v`, though there may be more

### CODE REVIEW

File: src/share/cxx/cxx_f90_interface.cpp [hunk 1]
```diff
@@ -170,6 +170,9 @@ void f90_push_forcing_to_cxx(F90Ptr elem_derived_FM, F90Ptr elem_derived_FT,
   sync_to_device(ft_f90, elements.m_ft);
 
   Tracers &tracers = Context::singleton().get_tracers();
+  if(tracers.fq.data() == nullptr) {
```
Comment (ambrad) [inline src/share/cxx/cxx_f90_interface.cpp]: I believe here you should allocate fq (and copy the fq data) only if the ftype requires it. As the code in this PR is currently written, I believe the first time any type of forcing is done, fq will get allocated. We want to allocate fq only if the forcing type requires it. fq is a an array whose size scales with QSIZE_D, which makes it the largest kind of array that gets allocated in Hommexx.

Comment (mfdeakin-sandia) [reply src/share/cxx/cxx_f90_interface.cpp]: I expected as much, I'll be making the change tomorrow

File: src/share/element_mod.F90 [hunk 2]
```diff
@@ -835,9 +835,7 @@ subroutine setup_element_pointers(elem)
     allocate( elem_derived_dp               (np,np,nlev,nelemd)                    )
     allocate( elem_derived_divdp            (np,np,nlev,nelemd)                    )
     allocate( elem_derived_divdp_proj       (np,np,nlev,nelemd)                    )
-#ifndef USE_KOKKOS_KERNELS
     allocate( elem_derived_FQ               (np,np,nlev,qsize_d,nelemd) )
```
Comment (oksanaguba) [inline src/share/element_mod.F90]: could this have been affecting performance of hommexx in a better way? would not having this allocated make hommexx faster?

**Comment by mfdeakin-sandia** [comment]
```
*** The MPI_Request_free() function was called after MPI_FINALIZE was invoked.
*** This is disallowed by the MPI standard.
*** Your MPI job will now abort.
[node05.blake.sandia.gov:19579] Local abort after MPI_FINALIZE started completed successfully, but am not able to aggregate error messages, and not able to guarantee that all other processes were killed!
```
With the previous commit, we now seem to make it to the end before crashing as above.

**Comment by mfdeakin-sandia** [comment]
Oh yeah, this also passes tests on blake

**Pull request merged**